SentinelOne/sentinelone_module/logs/connector.py

```diff
@@ -0,0 +1,132 @@
+from functools import cached_property
+import time
+
+from datetime import datetime, timedelta, timezone
+from prometheus_client import Counter, Histogram
+from management.mgmtsdk_v2.exceptions import UnauthorizedException
+from management.mgmtsdk_v2.services.activity import ActivitiesFilter
+from management.mgmtsdk_v2.services.threat import ThreatQueryFilter
+from management.mgmtsdk_v2_1.mgmt import Management
+from sekoia_automation.connector import Connector
+
+from sentinelone_module.logs.configuration import SentinelOneLogsConnectorConfiguration
+
+SENTINEL_ONE_HOW_MANY_DAYS_OF_EXISTING_LOGS_TO_RETRIEVE = 1
+
+# Declare prometheus metrics
+prom_namespace = "sicconfapi_intakes"
+
+OUTCOMING_EVENTS = Counter(
+    name="forwarded_events",
+    documentation="Number of events forwarded to SEKOIA.IO",
+    namespace=prom_namespace,
+    labelnames=["datasource", "intake_key"],
+)
+
+FORWARD_EVENTS_DURATION = Histogram(
+    name="forward_events_duration",
+    documentation="Duration to collect and forward events",
+    namespace=prom_namespace,
+    labelnames=["datasource", "intake_key"],
+)
```

Comment:
Can you outsource these declarations in a [`metric.py`](https://github.com/SEKOIA-IO/automation-library/blob/main/Checkpoint/connectors/metrics.py) file?
Also the metrics `EVENTS_LAG` is missing. Could you create it and use it with the difference between `now()` and the latest datetime seen in the events? Thanks

Reply:
I see that there is a `createdAt` field in the `Activity` or `Threat` classes  from SentinelOne.
Do I have an easy way to know how they will be formatted? So that I know e.g. should I check the diff between `now()` or `now(tz=timezone.utc)` or etc

Reply:
Added a method that computes this event lag based on the list of received events